core/storage/btree.rs

```diff
@@ -85,7 +85,7 @@ struct PageStack {
     /// cell_indices[current_page] is the current cell index being consumed. Similarly
     /// cell_indices[current_page-1] is the cell index of the parent of the current page
     /// that we save in case of going back up.
```

Comment:
Can you add comment saying negative indices means we've been going backwards?

Reply:
OK, here I want to use `-1` to indicate that the iteration has reached the beginning of the page. 
However, I feel that using `cell_idx = cell_count` to represent the end and `cell_idx = -1` represent the beginning of the page is somewhat confusing. It would be better if there was another way to express this. Do you have any ideas on how to handle this?

Reply:
maybe keep the logic abstract to a method with clear naming? or do you think even the implementation detail of it depending on cell idx being out of bounds is confusing?

Reply:
ok, added a method names `out_of_begin`